examples/vision/competitors.py

```diff
@@ -0,0 +1,257 @@
+import json
+import logging
+import os
+import sys
+from typing import Dict, List, Optional
+
+import instructor
+from dotenv import find_dotenv, load_dotenv
+from openai import OpenAI
+from pydantic import BaseModel, Field
+from rich import print as rprint
+
+load_dotenv(find_dotenv())
+
+# Add logger
+logging.basicConfig()
+logger = logging.getLogger("app")
+logger.setLevel("INFO")
+
+class Competitor(BaseModel):
+    name: str
+    features: Optional[List[str]]
+
+
+# Define models
+class Industry(BaseModel):
+    """
+    Represents competitors from a specific industry extracted from an image using AI.
+    """
+
+    name: str = Field(
+        description="The name of the industry"
+    )
+    competitor_list: List[Competitor] = Field(
+        description="A list of competitors for this industry"
+    )
+
+class Competition(BaseModel):
+    """
+    Represents competitors extracted from an image using AI.
+
+    This class serves as a structured representation of 
+    competitors and their qualities.
+    """
+
+    industry_list: List[Industry] = Field(
+        description="A list of industries and their competitors"
+    )
+
+# Define clients
+client_image = instructor.patch(
+    OpenAI(api_key=os.getenv("OPENAI_API_KEY")), mode=instructor.Mode.MD_JSON
+)
+
+# Define functions
+def read_images(image_urls: List[str]) -> Competition:
+    """
+    Given a list of image URLs, identify the competitors in the images.
+    """
+
+    logger.info(f"Identifying competitors in images... {len(image_urls)} images")
+
+    return client_image.chat.completions.create(
+        model="gpt-4-vision-preview",
+        response_model=Competition,
+        max_tokens=2048,
+        temperature=0,
+        messages=[
+            {
+                "role": "user",
+                "content": [
+                    {
+                        "type": "text",
+                        "text": "Identify competitors and generate key features for each competitor.",
+                    },
+                    *[
+                        {"type": "image_url", "image_url": {"url": url}}
+                        for url in image_urls
+                    ],
+                ],
+            }
+        ],
+    )
+
+
+
+def run(images: List[str]) -> Competition:
+    """
+    Given a list of images, identify the industries and the competitors in the images.
+    """
+
+    competitors: Competition = read_images(images)
+
+    return competitors
+
+
+if __name__ == "__main__":
+    # Run logger
```

Comment:
I don't think we need all this additional stuff
We can just have it be a variable, right? 
or maybe make it a typer cli?